### PR TITLE
feat(version): add support for monorepos

### LINKED ISSUE #1082: [QUESTION] npm version doesn't create git commit/tag in subpackages
# What / Why
When running `npm version major` or any other `npm version` command in sub-package (eg. monorepo with `packages/subpackage` folder), git commit and git tag is not created for the git repository, while the `package.json` and `package-lock.json` files are updated in a subpackage. 
Is this expected behavior? 

## Where
* n/a

## Who
* n/a

## References
* n/a

### PR DESCRIPTION
**Pull request by dhruvdutt**
The `.git` folder lookup in `version` command is based on localPrefix which is not the right location always for monorepos where `.git` folder is at a different location and not there `npm version` command is executed.

Fixes by doing the right lookup that will support `.git` in monorepos.

## References
Fixes #1082, [#9111](https://github.com/npm/npm/issues/9111)

### CODE REVIEW

File: lib/version.js [hunk 1]
```diff
@@ -210,7 +210,11 @@ function dump (data, cb) {
 }
 
 function statGitFolder (cb) {
-  fs.stat(path.join(npm.localPrefix, '.git'), cb)
+  git.whichAndExec(
+    [ 'rev-parse', '--git-dir' ],
+    { env: process.env },
+    cb
+  )
```
Comment (ljharb) [inline lib/version.js]: this doesn't respect `npm.localPrefix` tho. perhaps:
```suggestion
  git.whichAndExec(
    [ 'rev-parse', '--git-dir' ],
    { env: process.env, cwd: npm.localPrefix },
    cb
  )
```

Comment (dhruvdutt) [reply lib/version.js]: @ljharb Can you please understanding how to fix the test suite?

Comment (ljharb) [reply lib/version.js]: no, i'm afraid i have no idea

**Comment by dhruvdutt** [comment]
@ruyadorno Can you please guide on fixing tests?

**Comment by ruyadorno** [comment]
hi @dhruvdutt my understanding here is that tests are failing because they were relying on the fact that in the nested folder from which the commands are run (in the context of these tests) there is no `.git` folder, that would cause the command to fail, which is the expected behavior being asserted in these current failing tests: https://github.com/npm/cli/blob/latest/test/tap/version-lifecycle.js#L62

Since the `git rev-parse` command will work from any sub-directory of a git repo, the proposed change from this PR will result in these assertions failing - since the `statGitFolder` call will now succeed instead of failing as it was the case before, because it's reading rev-parse info from **the repo itself** 😁 

Now some important considerations: It's worth noting that we're heading towards end of beta for npm7 and this command has been entirely rewrote for the new version and this logic now lives on: https://github.com/npm/libnpmversion - so maybe a more long-lasting contribution would be to validate how it's currently behaving in the v7 beta and contribute the fixes to `libnpmversion` instead? (IMO this fix is actually  very important in v7 since that's the first version which will ship with support to [**workspaces**](https://github.com/npm/rfcs/blob/566d707e5da108b3fd38f5918d4e5271047d787b/accepted/0026-workspaces.md) but I'm not sure if it's still behaves the same in v7)

**Pull request closed without merging**